sequencer.c

```diff
@@ -986,10 +1000,18 @@ static int run_git_commit(struct repository *r,
 
 		if (res <= 0)
 			res = error_errno(_("could not read '%s'"), defmsg);
-		else
+		else {
+			if (opts->committer_date_is_author_date) {
+				char *date = read_author_date_or_null();
+				if (date)
```

Comment:
If the user requested the committer date to be set equal to the author date, but we for some reason can't read the author date, wouldn't it be better to fail hard, rather than ignoring the users wish?  Same in the other cases.

Reply:
Hmm, we can. And while on it, lets's change the function name to `read_author_date_or_die`.

Reply:
Yes, we can. And while on it, Lets change the function name to `read_author_date_or_die`.